core/shared/src/main/scala/zio/Task.scala

```diff
@@ -421,6 +421,11 @@ object Task extends TaskPlatformSpecific {
   def fromTry[A](value: => scala.util.Try[A]): Task[A] =
     ZIO.fromTry(value)
 
+  /**
+   * Wraps the value of the given option into a Task, if the option is empty the task fails with NoSuchElementException
+   */
+  final def getOrFail[A](v: => Option[A]): Task[A] = Task(v.get)
```

Comment:
I would move this implementation to the ZIO companion object and just alias it here.

Also, I think it will be slightly more efficient if you do something like `ZIO.effectSuspendTotal(v.fold(ZIO.failNow(new NoSuchElementException("None.get")))(ZIO.succeedNow))`. There is some cost in throwing an exception and creating the stack trace and we don't need that since we have our own trace. Also I think we want to make sure we die if any errors are thrown other than just the fact that the option is empty.